tests/mbt/driver/mbt_test.go

```diff
@@ -0,0 +1,577 @@
+
+	t.Log("Reading params...")
+	params := trace.States[0].VarValues["params"].Value.(itf.MapExprType)
+
+	consumersExpr := params["ConsumerChains"].Value.(itf.ListExprType)
+	initialValSetExpr := params["InitialValidatorSet"].Value.(itf.MapExprType)
+
+	initialValSet := make(map[string]int64)
+	for val, power := range initialValSetExpr {
+		initialValSet[val] = power.Value.(int64)
+	}
+
+	consumers := make([]string, len(consumersExpr))
+	for i, chain := range consumersExpr {
+		consumers[i] = chain.Value.(string)
+	}
+
+	chains := append(consumers, "provider")
+
+	t.Log("Chains are: ", chains)
+
+	// create params struct
+	vscTimeout := time.Duration(params["VscTimeout"].Value.(int64)) * time.Second
+
+	unbondingPeriodPerChain := make(map[ChainId]time.Duration, len(consumers))
+	trustingPeriodPerChain := make(map[ChainId]time.Duration, len(consumers))
+	ccvTimeoutPerChain := make(map[ChainId]time.Duration, len(consumers))
+	for _, consumer := range chains {
+		unbondingPeriodPerChain[ChainId(consumer)] = time.Duration(params["UnbondingPeriodPerChain"].Value.(itf.MapExprType)[consumer].Value.(int64)) * time.Second
+		trustingPeriodPerChain[ChainId(consumer)] = time.Duration(params["TrustingPeriodPerChain"].Value.(itf.MapExprType)[consumer].Value.(int64)) * time.Second
+		ccvTimeoutPerChain[ChainId(consumer)] = time.Duration(params["CcvTimeout"].Value.(itf.MapExprType)[consumer].Value.(int64)) * time.Second
+	}
+
+	modelParams := ModelParams{
+		VscTimeout:              vscTimeout,
+		CcvTimeout:              ccvTimeoutPerChain,
+		UnbondingPeriodPerChain: unbondingPeriodPerChain,
+		TrustingPeriodPerChain:  trustingPeriodPerChain,
+	}
+
+	valExprs := params["Nodes"].Value.(itf.ListExprType)
+	valNames := make([]string, len(valExprs))
+	for i, val := range valExprs {
+		valNames[i] = val.Value.(string)
+	}
+
+	// initialValSet has the right vals, but not yet the right powers
+	valSet, addressMap, signers, err := CreateValSet(initialValSet)
+	require.NoError(t, err, "Error creating validator set")
+
+	// get a slice of validators in the right order
+	nodes := make([]*cmttypes.Validator, len(valNames))
+	for i, valName := range valNames {
+		nodes[i] = addressMap[valName]
+	}
+
+	driver := newDriver(t, nodes, valNames)
+	driver.DriverStats = &stats
+
+	driver.setupProvider(modelParams, valSet, signers, nodes, valNames)
+
+	// remember the time offsets to be able to compare times to the model
+	// this is necessary because the system needs to do many steps to initialize the chains,
+	// which is abstracted away in the model
+	timeOffset := driver.runningTime("provider")
+
+	t.Log("Started chains")
+
+	t.Log("Reading the trace...")
+
+	for index, state := range trace.States {
+		t.Logf("Reading state %v of trace %v", index, path)
+
+		// modelState := state.VarValues["currentState"]
+		trace := state.VarValues["trace"].Value.(itf.ListExprType)
+		// fmt.Println(modelState)
+		lastAction := trace[len(trace)-1].Value.(itf.MapExprType)
```

Comment:
Might be useful to have a comment on why we only care of the `lastAction`. If my understanding is correct, subsequent `state`s always contain all the actions of all previous `state`s.